### ISSUE TITLE
Smart indentation is broken in Python

### ISSUE THREAD

**Issue by jkpubsrc** [microsoft/pylance-release#5741]
**VSCode:**

Version: 1.88.0-insider
Commit: d994aede3529f4d1af9eeaeb234d32fd936243e7
Date: 2024-03-29T22:23:13.365Z
Electron: 28.2.8
ElectronBuildId: 27744544
Chromium: 120.0.6099.291
Node.js: 18.18.2
V8: 12.0.267.19-electron.0
OS: Linux x64 5.4.0-169-generic

**Python-Extension:**

Published: 2016-01-19, 16:03:11
Last released: 2024-04-04, 21:07:36
Last updated: 2024-04-04, 21:20:21
Identifier: ms-python.python

**Description**

For some unexpected reasons automatic indentation is broken for comments.
Expected behavior: On pressing return you would expect the cursor to be placed at the first non-whitespace character of the previous line.
Encountered behavior: Instead of indenting with a single tab a single space is used.
(See below.)

**Steps to reproduce**

Settings:
* Editor: Insert Spaces is off
* Editor: Detect indentation is off
* Indentation is configured to insert tabs. Each tab is 4 characters in size.

Enter the following example code in a new python file:

```
def foo():
	a = 2
	# abc
	pass
```

Put the cursor to the right of "# abc". So the cursor is at the end of the line:

```
def foo():
	a = 2
	# abc_
	pass
```

(Here the underscore represents the cursor.)

Now press <Enter>.

Expected:

```
def foo():
	a = 2
	# abc
	# _
	pass
```

Encountered

```
def foo():
	a = 2
	# abc
 # _
	pass
```

(The underscore marks the position of the cursor after pressing <Enter>.)

This behavior is new somehow. Unfortunately I can't tell when I encountered it the first time but as far as I remember indentation worked correctly a few months ago.

This is quite annoying. You have to interrupt your current flow and correct the indentation every time you enter a comment.

**Comment by KacieKK** [comment]
Hey @jkpubsrc, thanks for reporting this bug. I tested it and can conform this be reproduced.

**Comment by debonte** [comment]
This issue has been fixed in prerelease version 2024.4.101, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#20244101-10-April-2024-PreRelease)
